lib/fastlane/plugin/sonarcloud_metric_kit/actions/sonarcloud_metric_kit_action.rb

```diff
@@ -0,0 +1,182 @@
+require 'fastlane/action'
+
+module Fastlane
+  module Actions
+    class SonarcloudMetricKitAction < Action
+      def self.run((params))
+        collect_metrics(params)
+      end
+
+      def self.api_url
+        'https://sonarcloud.io/api'
+      end
+
+      def self.collect_metrics(params)
+        metrics = {}
+        measures = measures(params)
+        metric_keys.each do |key|
+          metric = measures.find { |m| m['metric'] == key.to_s }
+          metrics[key.to_sym] =
+            if metric['value']
+              metric['value']
+            elsif params[:period] && metric['periods'].size > params[:period]
+              metric['periods'][params[:period]]['value']
+            else
+              metric['periods']
+            end
+        end
+        return metrics unless params[:quality_gate]
+
+        quality_gate = quality_gate(params)
+        metrics[:quality_gate] = quality_gate
+        if params[:period] && quality_gate['periods'].size > params[:period]
+          metrics[:period_value] = quality_gate['periods'][params[:period]]['parameter']
+        end
+        metrics
+      end
+
+      def self.measures(params)
+        queries = "componentKey=#{params[:project_key]}&metricKeys=#{metric_keys.join(',')}"
+        url = URI("#{api_url}/measures/component?#{queries}")
+        response = request(url, token: params[:sonar_token])
+        json_parse(response.read_body)['component']['measures']
+      end
+
+      def self.quality_gate(params)
+        url = URI("#{api_url}/qualitygates/project_status?projectKey=#{params[:project_key]}")
+        JSON.parse(request(url, token: params[:sonar_token]).read_body)['projectStatus']
+      end
```

Comment:
I think adding a prefix like "get" is useful to understand that this actually performs a HTTP Request, otherwise by reading the collect_metrics function, is not very clear that it is performance requests.
```suggestion
      def self.get_measures(params)
        queries = "componentKey=#{params[:project_key]}&metricKeys=#{metric_keys.join(',')}"
        url = URI("#{api_url}/measures/component?#{queries}")
        response = request(url, token: params[:sonar_token])
        json_parse(response.read_body)['component']['measures']
      end

      def self.get_quality_gate(params)
        url = URI("#{api_url}/qualitygates/project_status?projectKey=#{params[:project_key]}")
        JSON.parse(request(url, token: params[:sonar_token]).read_body)['projectStatus']
      end
```

Reply:
Done, thanks a mil 👍